cub/cub/thread/thread_operators.cuh

```diff
@@ -354,6 +354,64 @@ struct ReduceBySegmentOp
   }
 };
 
+template <typename ScanOpT>
+struct ScanBySegmentOp
```

Comment:
question: should we put it into detail namespace?

Reply:
Yes, makes sense. I moved it to the `detail` namespace.